Neither method after Connect and Login works

All these methods throw 230 error:

`ftp.Cwd("/")`
`ftp.Pwd()`
`ftp.List("/")`

I want to connect to IMDB ftp, `ftp://ftp.fu-berlin.de`. I'm trying to use your package.

Here is an example of the error:
```
$ read_db
List Error
panic: 230 FTP.FU-Berlin.DE login ok.


goroutine 1 [running]:
main.main()
        C:/Users/user/sbox/go/src/bitbucket.org/wzup/read_db/readDb.go:51 +0x47b
```

Here is my code, just a copy/paste of your example:

```
package main
import (
    "fmt"
    "github.com/dutchcoders/goftp"
)
func main() {

    /**
     * dutchcoders/goftp
     */
    var err error
    var ftp *goftp.FTP

    // For debug messages: goftp.ConnectDbg("ftp.server.com:21")
    if ftp, err = goftp.Connect("ftp.fu-berlin.de:21"); err != nil {
        fmt.Println("Connect Error")
        panic(err)
    }

    defer ftp.Close()

    if err = ftp.Login("admin", "sdkjcfnjoindsio"); err != nil {
        fmt.Println("Login Error")
        panic(err)
    }

    if err = ftp.Cwd("/"); err != nil {
        fmt.Println("Cwd Error")
        panic(err)
    }

    var curpath string
    if curpath, err = ftp.Pwd(); err != nil {
        fmt.Println("Pwd Error")
        panic(err)
    }
    fmt.Printf("Current path: %s", curpath)

    var files []string
    if files, err = ftp.List("/"); err != nil {
        fmt.Println("List Error")
        panic(err)
    }
    fmt.Println(files)
}
```

What do I do wrong? Why I cannot get an access to ftp server and see and download its contents?

--- a/ftp.go
+++ b/ftp.go
@@ -362,7 +362,7 @@ func (ftp *FTP) receiveLine() (string, error) {
 func (ftp *FTP) getCode(line string) (int, bool, error) {
 	fields := strings.Fields(strings.Trim(line, " \r\n"))
 	if len(fields) == 0 {
-		return -1, false, errNoCode
+		return 0, false, errNoCode
 	}
 
 	if len(fields[0]) == 3 {
@@ -374,7 +374,7 @@ func (ftp *FTP) getCode(line string) (int, bool, error) {
 			return code, true, nil
 		}
 	}
-	return -1, false, errNoCode
+	return 0, false, errNoCode
 }
 
 func (ftp *FTP) receive() (string, error) {

--- a/ftp.go
+++ b/ftp.go
@@ -360,7 +360,8 @@ func (ftp *FTP) receiveLine() (string, error) {
 }
 
 func (ftp *FTP) getCode(line string) (int, bool, error) {
-	fields := strings.Fields(strings.Trim(line, " \r\n"))
+	trimmedLine := strings.Trim(line, " \r\n")
+	fields := strings.Fields(trimmedLine)
 	if len(fields) == 0 {
 		return 0, false, errNoCode
 	}
@@ -372,6 +373,12 @@ func (ftp *FTP) getCode(line string) (int, bool, error) {
 	} else if len(fields[0]) == 4 && fields[0][3] == '-' {
 		if code, err := strconv.Atoi(fields[0][:3]); err == nil {
 			return code, true, nil
+		}
+	} else {
+		if len(trimmedLine) >= 4 && trimmedLine[3] == '-' {
+			if code, err := strconv.Atoi(trimmedLine[:3]); err == nil {
+				return code, true, nil
+			}
 		}
 	}
 	return 0, false, errNoCode

--- a/ftp.go
+++ b/ftp.go
@@ -15,6 +15,7 @@ import (
 )
 
 var REGEX_PWD_PATH *regexp.Regexp = regexp.MustCompile(`\"(.*)\"`)
+var errNoCode = errors.New("No code")
 
 type FTP struct {
 	conn net.Conn
@@ -355,8 +356,25 @@ func (ftp *FTP) receiveLine() (string, error) {
 	if ftp.debug {
 		log.Printf("< %s", line)
 	}
-
 	return line, err
+}
+
+func (ftp *FTP) getCode(line string) (int, bool, error) {
+	fields := strings.Fields(strings.Trim(line, " \r\n"))
+	if len(fields) == 0 {
+		return -1, false, errNoCode
+	}
+
+	if len(fields[0]) == 3 {
+		if code, err := strconv.Atoi(fields[0]); err == nil {
+			return code, false, nil
+		}
+	} else if len(fields[0]) == 4 && fields[0][3] == '-' {
+		if code, err := strconv.Atoi(fields[0][:3]); err == nil {
+			return code, true, nil
+		}
+	}
+	return -1, false, errNoCode
 }
 
 func (ftp *FTP) receive() (string, error) {
@@ -365,25 +383,21 @@ func (ftp *FTP) receive() (string, error) {
 	if err != nil {
 		return line, err
 	}
-
-	if (len(line) >= 4) && (line[3] == '-') {
+	code, beginMultiline, err := ftp.getCode(line)
+	if err == nil && beginMultiline {
 		//Multiline response
-		closingCode := line[:3] + " "
+		closingCode := code
 		for {
 			str, err := ftp.receiveLine()
 			line = line + str
 			if err != nil {
 				return line, err
 			}
-			if len(str) < 4 {
-				if ftp.debug {
-					log.Println("Uncorrectly terminated response")
-				}
+			code, _, err := ftp.getCode(str)
+			if err == nil && code == closingCode {
 				break
 			} else {
-				if str[:4] == closingCode {
-					break
-				}
+				//check error codes 5xx 4xx
 			}
 		}
 	}

--- a/ftp_test.go
+++ b/ftp_test.go
@@ -187,6 +187,8 @@ func TestGetCode(t *testing.T) {
 		{"220asdf test", getCodeResult{0, false, errNoCode}},
 		{"", getCodeResult{0, false, errNoCode}},
 		{"\r\n", getCodeResult{0, false, errNoCode}},
+		{"123-Firstline", getCodeResult{123, true, nil}},
+		{"123-First line", getCodeResult{123, true, nil}},
 	}
 	ftp := &FTP{}
 	for _, test := range tests {

--- a/ftp_test.go
+++ b/ftp_test.go
@@ -168,3 +168,32 @@ func TestGetFilesListOnGoodServer(t *testing.T) {
 
 	connection.Close()
 }
+
+type getCodeResult struct {
+	code           int
+	beginMultiline bool
+	err            error
+}
+
+func TestGetCode(t *testing.T) {
+	var tests = []struct {
+		input string
+		want  getCodeResult
+	}{
+		{"220 test", getCodeResult{220, false, nil}},
+		{"220     test", getCodeResult{220, false, nil}},
+		{"  220     test", getCodeResult{220, false, nil}},
+		{"220- test", getCodeResult{220, true, nil}},
+		{"220asdf test", getCodeResult{-1, false, errNoCode}},
+		{"", getCodeResult{-1, false, errNoCode}},
+		{"\r\n", getCodeResult{-1, false, errNoCode}},
+	}
+	ftp := &FTP{}
+	for _, test := range tests {
+		code, beginMultiline, err := ftp.getCode(test.input)
+		res := getCodeResult{code, beginMultiline, err}
+		if res != test.want {
+			t.Errorf("want: %#v, expected: %#v", test.want, res)
+		}
+	}
+}

--- a/ftp_test.go
+++ b/ftp_test.go
@@ -184,9 +184,9 @@ func TestGetCode(t *testing.T) {
 		{"220     test", getCodeResult{220, false, nil}},
 		{"  220     test", getCodeResult{220, false, nil}},
 		{"220- test", getCodeResult{220, true, nil}},
-		{"220asdf test", getCodeResult{-1, false, errNoCode}},
-		{"", getCodeResult{-1, false, errNoCode}},
-		{"\r\n", getCodeResult{-1, false, errNoCode}},
+		{"220asdf test", getCodeResult{0, false, errNoCode}},
+		{"", getCodeResult{0, false, errNoCode}},
+		{"\r\n", getCodeResult{0, false, errNoCode}},
 	}
 	ftp := &FTP{}
 	for _, test := range tests {

